File: 1396-count-servers-that-communicate/count-servers-that-communicate.cpp
class Solution {
public:
    int countServers(vector<vector<int>>& grid) {
        int r = grid.size(), c = grid[0].size();
        vector<int> row(r, 0), col(c, 0);
        int count = 0;

        for (int i = 0; i < r; i++) {
            for (int j = 0; j < c; j++) {
                if (grid[i][j] == 1) {
                    row[i]++;
                    col[j]++;
                }
            }
        }

        for (int i = 0; i < r; i++) {
            for (int j = 0; j < c; j++) {
                if (grid[i][j] == 1) {
                    if (row[i] > 1 || col[j] > 1) {
                        count++;
                    }
                }
            }
        }

        return count;
    }
};

File: 1396-count-servers-that-communicate/count-servers-that-communicate.py
class Solution:
    def countServers(self, grid: List[List[int]]) -> int:
        r = len(grid)
        c = len(grid[0])
        row = [0] * r
        col = [0] * c
        count = 0

        for i in range(r):
            for j in range(c):
                if grid[i][j] == 1:
                    row[i] += 1
                    col[j] += 1

        for i in range(r):
            for j in range(c):
                if grid[i][j] == 1:
                    if row[i] > 1 or col[j] > 1:
                        count += 1

        return count
